modules/tinymce/src/core/test/ts/browser/lookup/UserLookupTest.ts

```diff
@@ -0,0 +1,371 @@
+        );
+      }));
+    });
+
+    it('TINY-11974: Should handle large batches of users efficiently', async () => {
+      const editor = hook.editor();
+      const userIds = Arr.range(100, (i) => `test-user-${i}`);
+
+      const promises = editor.userLookup.fetchUsers(userIds);
+      const users = await Promise.all(promises);
+
+      await Promise.all(Arr.map(users, async (user, index) =>
+        expect(Promise.resolve(user)).to.eventually.have.property('id').that.equals(`test-user-${index}`)
+      ));
+    });
+
+    it('TINY-11974: Should handle various combinations of optional properties correctly', async () => {
+      const editor = hook.editor();
+      const testCases = [
+        {
+          input: { id: 'user-1' },
+          expected: { id: 'user-1' }
+        },
+        {
+          input: { id: 'user-2', name: 'John' },
+          expected: { id: 'user-2', name: 'John' }
+        },
+        {
+          input: {
+            id: 'user-3',
+            name: undefined,
+            avatar: null,
+            description: '',
+            custom: {}
+          },
+          expected: { id: 'user-3', description: '', custom: {}}
+        },
+        {
+          input: {
+            id: 'user-4',
+            name: 'Jane',
+            avatar: 'avatar.jpg',
+            description: null,
+            custom: { role: 'admin' }
+          },
+          expected: {
+            id: 'user-4',
+            name: 'Jane',
+            avatar: 'avatar.jpg',
+            custom: { role: 'admin' }
+          }
+        }
+      ];
+
+      // Override the fetch_users fn to return our test cases
+      editor.options.set('fetch_users', () => Promise.resolve(Arr.map(testCases, (c) => c.input)));
+
+      const userIds = Arr.map(testCases, (c) => c.input.id);
+      const promises = editor.userLookup.fetchUsers(userIds);
+
+      await expect(Promise.all(promises)).to.eventually.deep.equal(
+        Arr.map(testCases, (c) => c.expected),
+        'Should resolve with expected user objects'
+      );
+    });
+  });
+
+  context('fetchUsers with override', () => {
+    let editor: Editor;
+    let originalFetchUsers: (userIds: string[]) => Promise<User[]>;
+
+    beforeEach(() => {
+      editor = hook.editor();
+      originalFetchUsers = editor.options.get('fetch_users');
+    });
+
+    afterEach(() => {
+      editor.options.set('fetch_users', originalFetchUsers);
+    });
+
+    it('TINY-11974: Should handle custom user properties', async () => {
+      const editor = hook.editor();
+      const userId = 'test-user-custom';
+
+      const customUser = {
+        id: userId,
+        name: 'Test User',
+        avatar: 'test-avatar.png',
+        description: 'Test Description',
+        custom: {
+          role: 'admin',
+          department: 'IT'
+        }
+      };
+
+      // Override fetch_users for this test
+      editor.options.set('fetch_users', () => Promise.resolve([ customUser ]));
```

Comment:
Should we deep clone here to make sure that the user data is a new fresh object since that would be closer to reality now it's just the same object reference.

Reply:
For this specific test, I think we should inline both objects so there's no cloning involved. Having a reference to the input variable in the asserted content can lead to tests passing incorrectly.